switch node version on make run

0xrohitgarg opened the pull request.

ashish-spext commented:
Failing Scenario: If the user has nvm installed but hasn't installed Node.js version 22.8.0 through setup.sh and prefers to use their current version.

Questions:
- What is the minimum Node.js version supported by our frontend?
- Is there any specific advantage to pinning the version to 22.8.0?

Exploration:
- Can we modify the command to allow users to either use version 22.8.0 or any version greater than the minimum supported version, to offer more flexibility?

0xrohitgarg commented:
Updated the makefile run command to run the following logic,

Two node version are mentioned in the script : 
- `required version` is the version that project installed during `setup.sh` which is current `lts` 
- `minimum version` is the minimum version required to run the project

Updated Logic 
- If nvm is installed
   - If `required version` is installed with nvm; use that
   - If `required version` is not present, then check versions above `minimum version`; use that
   - If no version is found, install the `required version` with nvm and use that
- If nvm is not installed
   - Use user's default node version

ashish-spext reviewed (approved):
All 3 cases are working.

1. Required version is available (use it)
2. `> min required version` is available (use it)
3. `< min required version` is available (install required version and use it)
Note: If nvm is installed it will need atleast one version so there won't be a case for no version is installed.

I think 3rd case might be super rare but it is good that it is covered.
There can be one more possibility in case 3, that npm install is not done. In that case user will get error `vite: not found` and they should be able to take care of installation.

Let's merge it after updating the title.
